documentation/public/user-guide/05-syntax-trees/01-parsing-source-code/examples/03-handling-syntax-errors.test.mts

```diff
@@ -0,0 +1,29 @@
+import assert from "node:assert";
+import { Parser } from "@nomicfoundation/slang/parser";
+import { assertIsTerminalNode, EdgeLabel, NonterminalKind, TerminalKind } from "@nomicfoundation/slang/cst";
+
+test("handling syntax errors", () => {
```

Comment:
TODO: need to explain a bit more about our error recovery strategy, and how we expect the produced trees to always be complete, regardless of the "unrecognized" section...

Also, need to differentiate in the examples between "unrecognized" and "missing" cases...